src/main/java/org/mifos/pheeBillPay/api/implementation/BillPaymentsController.java

```diff
@@ -20,6 +20,11 @@ public class BillPaymentsController implements BillPaymentsApi {
     public ResponseEntity<BillInquiryResponseDTO> billPayments(String tenantId, String correlationId, String callbackURL, String payerFspId,
             BillPaymentsReqDTO body) throws ExecutionException {
         BillInquiryResponseDTO billInquiryResponseDTO = new BillInquiryResponseDTO();
+        if(!checkforValidity(tenantId,correlationId,callbackURL,payerFspId,body).equals("Valid")){
+            billInquiryResponseDTO.setTransactionId("Invalid Request: Mandatory Fields Missing, Missing field is "+
+                    checkforValidity(tenantId,correlationId,callbackURL,payerFspId,body));
```

Comment:
Here calling this method(checkforValidity) two times is not needed. we can store the result in a temporary variable and reuse.